Mathlib/CategoryTheory/Limits/Shapes/Reflexive.lean

```diff
@@ -148,6 +171,491 @@ instance (priority := 100) hasCoreflexiveEqualizers_of_hasEqualizers [HasEqualiz
+                     | leftCompReflexion => dsimp
+                                            simp only [Functor.map_comp, obj_one, obj_zero,
+                                              map_right, map_reflexion, Category.assoc, h₃,
+                                              reassoc_of% h₁]
+                     | rightCompReflexion => dsimp
+                                             simp only [Functor.map_comp, obj_one, obj_zero,
+                                               map_left, map_reflexion, Category.assoc, h₃,
+                                               reassoc_of% h₂]
+
+/-- A version of `mkNatTrans` for natural transforms from a `reflexivePair` rather than to a
+`reflexivePair`. -/
+@[simps]
+def mkNatTrans' (u : A ⟶ F.obj one) (v : B ⟶ F.obj zero)
+    (h₁ : u ≫ (F.map left) = f ≫ v := by aesop_cat)
+    (h₂ : u ≫ (F.map right) = g ≫ v := by aesop_cat)
+    (h₃ : s ≫ u = v ≫ (F.map reflexion)  := by aesop_cat) :
+    (reflexivePair f g s sl sr) ⟶ F where
+  app := fun x ↦ match x with
+    | zero => v
+    | one => u
+  naturality := by intro _ _ f
+                   cases f with
+                     | id => simp
+                     | left => simp [h₁]
+                     | right => simp [h₂]
+                     | reflexion => simp [h₃]
+                     | leftCompReflexion => dsimp
+                                            simp only [Functor.map_comp, obj_one, obj_zero,
+                                              map_right, map_reflexion, Category.assoc, h₃,
+                                              reassoc_of% h₁]
+                     | rightCompReflexion => dsimp
+                                             simp only [Functor.map_comp, obj_one, obj_zero,
+                                               map_left, map_reflexion, Category.assoc, h₃,
+                                               reassoc_of% h₂]
+
+/-- Constructor for natural isomorphisms with a `reflexivePair`. -/
+@[simps!]
+def mkNatIso (u : F.obj one ≅ A) (v : F.obj zero ≅ B)
+    (h₁ : (F.map left) ≫ v.hom = u.hom ≫ f := by aesop_cat)
+    (h₂ : (F.map right) ≫ v.hom = u.hom ≫ g := by aesop_cat)
+    (h₃ : (F.map reflexion) ≫ u.hom = v.hom ≫ s := by aesop_cat) :
+    F ≅ (reflexivePair f g s sl sr) :=
+  NatIso.ofComponents (fun x ↦ match x with
+    | zero => v
+    | one => u)
+  (by intro _ _ f
+      cases f with
+        | id => simp
+        | left => simp [h₁]
+        | right => simp [h₂]
+        | reflexion => simp [h₃]
+        | leftCompReflexion => dsimp
+                               simp only [Functor.map_comp, obj_one, obj_zero,
+                                map_right, map_reflexion, Category.assoc, h₃,
+                                reassoc_of% h₁]
+        | rightCompReflexion => dsimp
+                                simp only [Functor.map_comp, obj_one, obj_zero,
+                                 map_left, map_reflexion, Category.assoc, h₃,
+                                 reassoc_of% h₂])
```

Comment:
Arguably, this would be slightly better:
```suggestion
@[simps!]
/-- Constructor for natural isomorphisms with a `reflexivePair`. -/
@[simps]
def mkNatIso (u : F.obj one ≅ A) (v : F.obj zero ≅ B)
    (h₁ : (F.map left) ≫ v.hom = u.hom ≫ f := by aesop_cat)
    (h₂ : (F.map right) ≫ v.hom = u.hom ≫ g := by aesop_cat)
    (h₃ : (F.map reflexion) ≫ u.hom = v.hom ≫ s := by aesop_cat) :
    F ≅ reflexivePair f g s sl sr where
  hom := mkNatTrans u.hom v.hom 
  inv := mkNatTrans' u.inv v.inv
    (by rw [← cancel_epi u.hom, u.hom_inv_id_assoc, ← reassoc_of% h₁, v.hom_inv_id,
            Category.comp_id])
    (by rw [← cancel_epi u.hom, u.hom_inv_id_assoc, ← reassoc_of% h₂, v.hom_inv_id,
            Category.comp_id])
    (by rw [← cancel_epi v.hom, v.hom_inv_id_assoc, ← reassoc_of% h₃, u.hom_inv_id,
            Category.comp_id])
```